packages/dcos-net/extra/dcos-net-setup.py

```diff
@@ -70,8 +74,28 @@ def add_networkd_config(src):
         result.returncode = 0
         return result
 
+    # Restart networkd only if the configuration is updated
+    mtime = os.path.getmtime(dst)
+    result = subprocess.run(['systemctl', 'show', '--value',
```

Comment:
In which cases you expect this command to have `returncode != 1`?

I ran this on Fedora 29, and all those commands result in statuscode 0:
```sh
> systemctl show --value --property ActiveEnterTimestamp docker
Mon 2018-11-05 04:35:54 CET
> echo $?
0
> systemctl show --value --property ActiveEnterTimestamp bananas

> echo $?
0
> systemctl show --value --property NO_PROPERTY docker
> echo $?
0
>
```

Does that means that you can end up passing `empty string` to strptime, and run into an exception?

Reply:
+1

Reply:
Exception is exactly that behavior that I would expect in that case. It's the best that we can do here. Let me explain why.

`ActiveEnterTimestamp` is a well-known documented property, all active systemd services have it. Right before `systemctl show` the script checks that `systemd-networkd.service` is active.

Non-zero exit code of `systemctl show` is an expected case, i.e. `--value` argument is not supported in very old systemd versions, it can fail if there is an issue with dbus, etc. In that case we will just try to restart networkd.

If `systemd-networkd.service` is active and `systemctl show --value --property ActiveEnterTimestamp systemd-networkd.service` exists with successful exit code and returns empty string, there is nothing much we can do and we have to panic. It's not expected behavior. We can't ignore the result, we can catch the exception and print a human-readable error message, but I don't see what kind of problem we are solving by that.

Reply:
Thanks for the clarification @urbanserj.

Reply:
>  but I don't see what kind of problem we are solving by that.

This helps debugging.

Instead of rediscovering what you just described, a clear error message would allow the operator/support/us to quickly narrow down the problem. You also provided a really good description - can we please record it in the source code?  We can just copy-paste it as a comment.

These two things will definatelly help people working on this code base in the future.